### PR TITLE
Bump Deps, add Ord and From<&[u8]> to uint.

### PR DESCRIPTION
**Pull request by tomusdrw**

### CODE REVIEW

File: src/types/uint.rs [hunk 1]
```diff
@@ -93,6 +115,16 @@ macro_rules! impl_uint {
       }
     }
 
+    impl<'a> From<&'a [u8]> for $name {
```
Comment (rphmeier) [inline src/types/uint.rs]: `T: AsRef<[u8]>` might be even better

File: src/types/uint.rs [hunk 2]
```diff
@@ -93,6 +115,16 @@ macro_rules! impl_uint {
...
+      fn from(num: &'a [u8]) -> Self {
+        let mut arr = [0; $len];
+        for i in 0..cmp::min(num.len(), $len) {
+          arr[$len - 1 - i] = num[num.len() - 1 - i]
```
Comment (rphmeier) [inline src/types/uint.rs]: seems like this whole thing can be replaced with a `arr[..].copy_from_slice(&x[..len])` which will do a direct `memcpy` and not rely on LLVM optimizing the loop

**Comment by rphmeier** [comment]
LGTM modulo nits; `AsRef` isn't strictly required but I think it makes sense here.

File: src/types/uint.rs (hunk 1 above)
Comment (tomusdrw) [inline src/types/uint.rs]: Seems that it conflicts with `From<u64>` implementation :/

File: src/types/uint.rs (hunk 2 above)
Comment (tomusdrw) [inline src/types/uint.rs]: Fixed!

File: src/types/uint.rs (hunk 1 above)
Comment (rphmeier) [inline src/types/uint.rs]: Hm, I guess since `AsRef<[u8]>` _could_ be added to `u64` eventually :(
We'll have to wait for specialization then.

**Pull request merged**